docs/contributing/3.0-development/app-layout.rst

```diff
@@ -0,0 +1,241 @@
+  creation of circular imports.
+* Imports done inside the module's ``__init__.py`` should be relative and explict, e.g.
+  ``from .submodule import identifier1, identifier2``, NOT
+  ``from submodule import identifier1, identifier2`` and NOT
+  ``from .submodule import *``.
+
+Any module in ``pulp.app`` may be broken up in this way, such as
+:mod:`pulp.app.serializers` or :mod:`pulp.app.viewsets`, should do so in such a way
+that renders the implementation invisible to anyone importing from that module.
+
+pulp.app
+--------
+
+pulp.app is the package containing the core Pulp Platform Django application.
+This package contains all of the Pulp Platform models, serializers, and
+viewsets required to assemble Pulp's REST API and underlying database.
+
+pulp.app.apps
+^^^^^^^^^^^^^
+
+This module defines the base class used by all Pulp plugins to identify them to
+the Pulp Platform as plugins to be imported by Pulp, as well as the Pulp Platform
+application config class itself, which is exposed to plugin writers via the Plugin API.
+
+pulp.app.settings
+^^^^^^^^^^^^^^^^^
+
+This is the main settings module for the platform Django project, which puts together
+all of the various Django applications that the Pulp Platform depends on to function,
+as well as the Pulp Platform application itself and its plugins.
+
+Many things are defined in here, including the database settings, logging configuration,
+REST API settings, etc. This file also finds and registers Pulp plugins with the Pulp
+Platform Django Project, using the ``pulp.plugin`` entry point.
+
+In order to use django-related tools with the Pulp Platform, the platform must be installed,
+and the ``DJANGO_SETTINGS_MODULE`` environment var must be set to
+:mod:`pulp.app.settings`.
+
+pulp.app.urls
+^^^^^^^^^^^^^
+
+This module lists all API endpoints exposed by the Pulp REST API, including all viewsets
+registered with the API router, and all non-API views (should we ever write any).
+
+pulp.app.pagination
+^^^^^^^^^^^^^^^^^^^
+
+API-related pagination classes are defined here. The number of paginators defined in this file
+should be kept as small as possible to ensure consistent pagination across all API views.
+
+
+pulp.app.models
+^^^^^^^^^^^^^^^
+
+All models are contained in :mod:`pulp.app.models`.
+
+The Platform models are all importable directly from the ``pulp.app.models``
+namespace. All Pulp models should subclass :mod:`pulp.app.models.Model`, or
+one of its subclasses.
+
+.. note::
+
+    All models must exist in the pulp.app.models namespace in order to be
+    recognized by Django and included in the Django ORM.
+
+Master/Detail Models
+********************
+
+A few Pulp Platform models, including the Content model as well as
+Importers and Publishers, implement a strategy we refer to as "Master/Detail".
+The Master/Detail strategy, as implemented in Pulp, allows us to capture
+necessary relationships across a single master Model, while still allowing
```

Comment:
It means exactly what it means. I can't clarify it further or reword it without a more specific question.